### PR TITLE
improved title sort

### LINKED ISSUE #425: Title Sort
Remove numbers, special characters and articles from beginning of title sort.

### PR DESCRIPTION
**Pull request by mccalluc**
Fixes #425, I think.

Remember that title sort is complicated because we're actually sorting by all the titles: On a smaller results list this can lead to some counter-intuitive results. But for the particular problem described in the bug, this works:

[solr admin](http://localhost:8983/solr/blacklight-core/select?q=*%3A*&sort=title+asc&rows=50&fl=id%2Ctitle&wt=json&indent=true):
...
```json
      {
        "id":"cpb-aacip_155-17crjgpf",
        "title":"World Cafe -- Larry Kane On John Lennon 2005"},
      {
        "id":"cpb-aacip_155-859cp21j",
        "title":"World Cafe -- 1997-01-20 Sat/Mon -- Martin Luther King, Jr. 1997"},
```

### CODE REVIEW

**Comment by mccalluc** [comment]
More commits coming in. For reference, analysis in solr only touches the indexed value, not the stored value... but there is an admin tool for checking your analysis config:

[analysis admin](http://localhost:8983/solr/#/blacklight-core/analysis?analysis.fieldvalue=%20%20%20A%20Title%20with%20padding%20and%20numbers%20%28123%29%20and%20a%C3%A7%C3%A7ents&analysis.fieldtype=sort&verbose_output=1):
```
in:  A Title with padding and numbers (123) and aççents
out: title with padding and numbers (zzzz000123) and accents
```

File: jetty/solr/blacklight-core/conf/schema.xml [hunk 1]
```diff
@@ -455,6 +455,26 @@
...
+            <filter class="solr.PatternReplaceFilterFactory"
+                    pattern="(\d+)" replacement="zzzz$1" replace="all"/>
+            <!-- Zero pad to mask lexicographic sort behavior. -->
+            <filter class="solr.PatternReplaceFilterFactory"
+                    pattern="(\d+)" replacement="00000$1" replace="all"/>
+            <filter class="solr.PatternReplaceFilterFactory"
+                    pattern="0*([0-9]{6,})" replacement="$1" replace="all"/>
+        </analyzer>
+    </fieldType>
  </types>
```
Comment (afred) [inline jetty/solr/blacklight-core/conf/schema.xml]: did you find ^^ in a how-to online somewhere? I'd like to bookmark it

Comment (mccalluc) [inline jetty/solr/blacklight-core/conf/schema.xml]: Different parts of it from different places. The zero-padding came from http://www.andornot.com/blog/post/Make-numbers-behave-when-sorting-alphanumerically-in-Solr.aspx

Comment (afred) [inline jetty/solr/blacklight-core/conf/schema.xml]: will this correctly sort...

`title 1`
`title 10`
`title 2`

Comment (afred) [inline jetty/solr/blacklight-core/conf/schema.xml]: @mccalluc I tried to check out the `better-title-sort` so as to test the above case. Having trouble seeing how to do that. Also, tests are [failing locally](https://gist.github.com/afred/c2733171fcd2433e69ed). Thoughts?

**Comment by mccalluc** [comment]
Latest commits exercise the sorting logic better.

For your test failure, make sure you've either restarted jetty, or reloaded the schema from the solr webconsole.

**Comment by afred** [comment]
thanks!

**Pull request merged**